python/pysrc/kaskada/_timestream.py

```diff
@@ -88,18 +108,25 @@ def _call(
             TypeError: If the argument types are invalid for the given function.
             ValueError: If the argument values are invalid for the given function.
         """
+        # Determine the session. This is unfortunately necessary for creating literals
+        # and other expressions.
         if session is None:
-            session = next(
-                arg._ffi_expr.session() for arg in args if isinstance(arg, Timestream)
-            )
-
-        def make_arg(arg: Union[Timestream, Literal]) -> _ffi.Expr:
-            if isinstance(arg, Timestream):
-                return arg._ffi_expr
+            if input is None:
+                # If there is no input, then at least one of the argumets must be a
+                # Timestream. Specifically, we shouldn't have all literals, nor should
+                # we have all Callables that produce Timestreams.
+                session = next(
+                    arg._ffi_expr.session()
+                    for arg in args
+                    if isinstance(arg, Timestream)
+                )
```

Comment:
Should this transform the `StopIteration` failure (if no args are timestreams) to a more user-friendly message?